### PR TITLE
Chapter 10, Exercise 09 + restatement of Exercise 02

### PR DESCRIPTION
**Pull request by archiebrowne**

### CODE REVIEW

File: M1FExplained/Chapter10/Exercises/Exercise09.lean [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+lemma helper (a b n : ℤ) (hb : 0 < b) (hab : ∃ (s t : ℤ), s * a + t * b = n) : 
+    ∃ (s' t' : ℤ), 0 < s' ∧ s' * a + t' * b = n := by 
+  rcases hab with ⟨s, t, hst⟩
+  set p := (b - s) / b with hp
+  use s + p * b, t - p * a
+  constructor
+  · rw [hp, ← neg_lt_iff_pos_add']
+    calc
+      -s = (b - s) - b := by norm_num
```
Comment (kbuzzard) [inline M1FExplained/Chapter10/Exercises/Exercise09.lean]: `norm_num` is a tactic to simplify expressions involving *numerals* (e.g. 0, 1, 37) so is not the tactic to use here. `norm_num` tries `simp` during its algorithm and my guess is that it's `simp` which is solving this.

File: M1FExplained/Chapter10/Exercises/Exercise09.lean [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+lemma exercise09 (a b : ℤ) (hb : b > 0) (hab : IsCoprime a b) : 
+  ∀ (n : ℤ), ∃ (s t : ℤ), 0 < s ∧ s * a + t * b = n := by
+  rcases hab with ⟨s', t', hst⟩
+  intro n
+  have hab : ∃ (s t : ℤ), s * a + t * b = n := by
+    · use n * s', n * t'
+      calc
+        _ = n * (s' * a + t' * b) := by ring
+        _ = n := by rw [hst]; norm_num
```
Comment (kbuzzard) [inline M1FExplained/Chapter10/Exercises/Exercise09.lean]: again this is not a job for `norm_num`: my guess is that the goal is `n * 1 = n` and that this is being solved by `simp`. `norm_num` cannot do anything about equations which have variables in.

**Review by kbuzzard — COMMENTED** [review]
This looks fine to me but you might want to remove the `norm_num`s and try `simp` instead because this is surely what's going on

**Comment by archiebrowne** [comment]
changes pushed

**Comment by kbuzzard** [comment]
Can you fix the conflict?

**Comment by kbuzzard** [comment]
Thanks!

**Pull request merged**